### DISCUSSION TITLE
useForm: get type of values after transform

### DISCUSSION THREAD

**Discussion by noahsb** [mantinedev/mantine#3086] [Q&A] [answered]
Hi, thanks for all your hard work maintaining mantine!

I have a situation (similar to an example in the docs) where my forms initial values are different from what is submitted:

```
const form = useForm({
  initialValues: {
    comment: ''
  },

  transformValues: (values) => ({
    comment: values.comment.length > 0 > values.comment : undefined
  })
 });
```

There are many different use cases other than what I provided, but this one is handy so nullable/optional values aren't passed to the mutation. Normally, I use:

```
const handleSubmit(values: typeof form.values) => {
  // do stuff
}

<form onSubmit={form.onSubmit((values) => handleSubmit(values))}>
```

But because form.values only points to `initialValues`, the type is incorrect. I could write my own interface for this and force `form` to have the correct type, but what about when a forms initial values have different fields than the forms return? I would like to avoid writing two interfaces for a single form if possible. 

Is there a mechanism to get the forms submit type?

**Comment by rtivital** [comment, marked as answer]
I cannot think of any good way of providing types for transform values, so you should create a separate type for that on your side.

**Reply by BrendanC23** [reply]
I would love to be able to provide both an initial and final type to useForm. Similar to what happens in [React Hook Form](https://github.com/react-hook-form/react-hook-form/blob/5f3df1227c381a9127ab27f7d1d98a9cefee33d5/src/useForm.ts#L49C3-L49C66). These types can be obtained by using Zod's input and infer methods.

One use of this would be to have a form with a NumberInput field.  The initial value for the number input is `""`, but the value upon submission must be a number.
